test/unit/events.py

```diff
@@ -0,0 +1,313 @@
+                        "Value": "the_value_2"
+                    }
+                ],
+                "Timestamp": 1528236844480,
+                "Value":123,
+                "Unit": "Seconds",
+                "StorageResolution": 12
+            }
+        ]
+    }
+
+@pytest.fixture()
+def StatisticValues_missing_Sum():
+    return {
+        "request_id": "the_id",
+        "metric_data": [
+            {
+                "MetricName": "the_metric_name",
+                "Dimensions": [
+                    {
+                        "Name": "the_name_1",
+                        "Value": "the_value_1"
+                    },
+                    {
+                        "Name": "the_name_2",
+                        "Value": "the_value_2"
+                    }
+                ],
+                "Timestamp": 1528236844480,
+                "StatisticValues": {
+                    "SampleCount": 12.17,
+                    "Minimum": 12.17,
+                    "Maximum": 12.17
+                },
+                "Unit": "Seconds",
+                "StorageResolution": 12
+            }
+        ]
+    }
+
+@pytest.fixture()
+def Unit_type_not_available():
+    return {
+        "request_id": "the_id",
+        "metric_data": [
+            {
+                "MetricName": "the_metric_name",
+                "Dimensions": [
+                    {
+                        "Name": "the_name_1",
+                        "Value": "the_value_1"
+                    },
+                    {
+                        "Name": "the_name_2",
+                        "Value": "the_value_2"
+                    }
+                ],
+                "Timestamp": 1528236844480,
+                "StatisticValues": {
+                    "SampleCount": 12.17,
+                    "Sum": 12.17,
+                    "Minimum": 12.17,
+                    "Maximum": 12.17
+                },
+                "Unit": "minutes",
+                "StorageResolution": 12
+            }
+        ]
+    }
+
+@pytest.fixture()
+def StorageResolution_type_invalid():
+    return {
+        "request_id": "the_id",
+        "metric_data": [
+            {
+                "MetricName": "the_metric_name",
+                "Dimensions": [
+                    {
+                        "Name": "the_name_1",
+                        "Value": "the_value_1"
+                    },
+                    {
+                        "Name": "the_name_2",
+                        "Value": "the_value_2"
+                    }
+                ],
+                "Timestamp": 1528236844480,
+                "StatisticValues": {
+                    "SampleCount": 12.17,
+                    "Sum": 12.17,
+                    "Minimum": 12.17,
+                    "Maximum": 12.17
+                },
+                "Unit": "Seconds",
+                "StorageResolution": "12"
+            }
+        ]
+    }
+
+@pytest.fixture()
+def Dimension_type_invalid():
```

Comment:
Dimensions invalid cases include: 
1. not an array
2. wrong type of item in the array
3. wrong property of item

Reply:
Dimension_type_invalid deals with 1. Will add separate tests for 2 and 3.